src/src/services/aws-rekognition/aws-rekognition.service.ts

```diff
@@ -0,0 +1,370 @@
+			imageName,
+		);*/
+		const response = { success: false, faceId: null };
+		try {
+			const addFaceParams = {
+				CollectionId: collectionId,
+				Image: {
+					S3Object: {
+						Bucket: this.bucketName,
+						Name: imageName,
+					},
+				},
+				ExternalImageId: imageName,
+				MaxFaces: 1,
+			};
+			/*console.log(
+				`\nSTART - Add face into a collection with params:\n`,
+				addFaceParams,
+			);*/
+
+			const addFaceResponse = await this.rekognition.send(
+				new IndexFacesCommand(addFaceParams),
+			);
+
+			//console.log(`\nSTART - Add face into a collection. Success!\n`);
+			//console.dir(addFaceResponse, { depth: 99 });
+			if (addFaceResponse.FaceRecords.length === 1) {
+				response.success = true;
+				response.faceId = addFaceResponse.FaceRecords[0].Face.FaceId;
+			}
+			return response;
+		} catch (error) {
+			console.log(
+				`\n  END - Add face into a collection. Error!\n`,
+				error,
+				error.stack,
+			);
+			return response;
+		}
+	}
+
+	async associateFaceToUser(
+		collectionId: string,
+		userId: string,
+		faceId: string,
+	) {
+		const response = { success: false };
+		try {
+			const associateFacesParams = {
+				CollectionId: collectionId.toString(),
+				UserId: (this.prefixed + userId).toString(),
+				FaceIds: [faceId.toString()],
+				UserMatchThreshold: Number('80'),
+				ClientRequestToken: (
+					this.prefixed + new Date().getTime()
+				).toString(),
+			};
+			/*console.log(
+				'associateFacesParams------->>>>>>>',
+				associateFacesParams,
+			);*/
+
+			const associateFaceResponse = await this.rekognition.send(
+				new AssociateFacesCommand(associateFacesParams),
+			);
+			//.promise();
+			//console.log('associateFaceResponse:');
+			//console.dir(associateFaceResponse);
+			if (associateFaceResponse.AssociatedFaces.length === 1)
+				response.success = true;
+			return response;
+		} catch (error) {
+			console.log('associateFaceToUser:', error, error.stack);
+			return response;
+		}
+	}
+
+	async searchUsersByImage(
+		collectionId: string,
+		imageName: string,
+		faceMatchingThreshold: number,
+	) {
+		try {
+			const searchParams = {
+				CollectionId: collectionId,
+				Image: {
+					S3Object: {
+						Bucket: this.bucketName,
+						Name: imageName,
+					},
+				},
+				UserMatchThreshold: faceMatchingThreshold,
+				MaxUsers: 5,
+			};
+			//console.log('searchParams:------------->>>>', searchParams);
+			const compareResponse = await this.rekognition.send(
+				new SearchUsersByImageCommand(searchParams),
+			);
+			//.promise();
+			//console.log('Matching faces:');
+			//console.dir(compareResponse, { depth: 99 });
+			return compareResponse.UserMatches;
+		} catch (error) {
+			console.log('searchUsersByImage:', error, error.stack);
+			return [];
+		}
+	}
+
+	async deleteCollection(
+		collectionId: string,
+		userId: string,
+		faceId: string,
+	) {
+		try {
+			const deleteFaceParams = {
+				CollectionId: collectionId,
+				userId: this.prefixed + userId,
```

Comment:
1. rename `const deleteFaceParams to const deleteCollectionParams`

2. are these needed for deleting collection?

```
				userId: this.prefixed + userId,
				FaceIds: [faceId],
```

Reply:
userId: this.prefixed + userId,	FaceIds: [faceId], not required in DeleteCollectionCommand
I read it on below documentation link
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-rekognition/classes/deletecollectioncommand.html

Reply:
I renamed const deleteFaceParams to const deleteCollectionParams. Also Removed userId, FaceId from deleteCollection method